zingoconfig/src/lib.rs

```diff
@@ -350,66 +376,198 @@ impl std::fmt::Display for ChainType {
 
 use ChainType::*;
 impl Parameters for ChainType {
-    fn activation_height(
-        &self,
-        nu: NetworkUpgrade,
-    ) -> Option<zcash_primitives::consensus::BlockHeight> {
+    fn activation_height(&self, nu: NetworkUpgrade) -> Option<BlockHeight> {
         match self {
             Mainnet => MAIN_NETWORK.activation_height(nu),
             Testnet => TEST_NETWORK.activation_height(nu),
-            FakeMainnet | Regtest => Some(BlockHeight::from_u32(1)),
+            // This function should never be called in regtest but fails to send in integ tests if None??
+            Regtest => Some(BlockHeight::from_u32(1)),
```

Comment:
Could you move the `ActivationHeights` (or even the entire `RegtestNetwork`, if that doesn't break how it's currently used too much) inside of the the `ChainType::Regtest` enum variant so that we can use this method in regtest mode? transaction construction calls this function in order to determine what transaction format to create

Reply:
yeah great idea